contents/blog/clickhouse-vs-snowflake.md

```diff
@@ -0,0 +1,146 @@
+
+This time, let’s start with ClickHouse. ClickHouse has three significant optimizations that make querying aggregate computations efficient: (i) materialized views, (ii) specialized engines, and (iii) vectorized query execution.
+
+- **Materialized views:** Materialized views are independent tables derived from other table data generated at some specific point in time. Unlike databases like Postgres, ClickHouse's materialized views are constantly re-generated in the background after new data is inserted. While materialized views are delayed because most of ClickHouse's use cases involve aggregate, analytical data, the delay doesn’t pose a problem.
+
+- **Specialized engines:** Materialized views are aided by ClickHouse's specialized engines that efficiently parallelize processes. For instance, if you need to add millions of numbers, ClickHouse's SummingMergeTree would dramatically optimize that query.
+
+- **Vectorized query execution:** Vectorized execution, initially championed by the MonetDB team, batches data to achieve bulk processing. Vectorized query execution takes advantage of CPU caches and SIMD CPU instructions to enable faster computation.
+
+Snowflake has some of these features, but with caveats: 
+
+- While Snowflake has support for materialized views, it charges additionally for it since it needs to allocate serious CPU resources to recompute the views. Those re-computations, under the hood, are more resource intensive for Snowflake because it lacks the same partial-update techniques ClickHouse uses to expedite re-calculations.
+
+- **Virtual warehouses instead of specialized engines:** Unlike ClickHouse's specialized engines, which utilize hardware to optimize query execution, Snowflake segments its database into virtual warehouses sized from small to large. These warehouses can have larger or lesser compute to handle complex to easy queries and isolate workloads. However, while clever and organized, this approach is computationally and financially expensive when taken full advantage of.
+
+- **Search optimization service:** Introduced in 2021, Snowflake offers a search optimization service at its higher enterprise tiers. Because Snowflake is a closed-source product, not much is known about how the search optimization service works, but it can offer enormous performance gains (4x-100x) relative to un-optimized search for some types of queries. Unfortunately, while Snowflake likely uses similar mechanisms to ClickHouse, direct side-by-side analysis isn’t possible due to Snowflake’s closed-search tier. By extension, the search optimization service’s performance boost is unpredictable.
+
+Overall, ClickHouse's close-metal optimizations enable it to return aggregate values over a thousand times faster (and cheaper) than Snowflake. However, Snowflake’s approach is considerably more user-friendly as it requires less SQL-level optimization to take advantage of these organizational features.
+
+### Security
```

Comment:
I would be in favour of just removing this as it doesn't add much.

Reply:
Agreed